### ISSUE TITLE
False negative when sub-classing a `Generic[T]`

### ISSUE THREAD

**Issue by egeres** [microsoft/pyright#5776]
I hope I'm not mistaken, but the following code has a typing error not being caught my pyright:

```python
from abc import ABC, abstractmethod
from typing import Generic, TypeVar

T = TypeVar("T")

class Foo(ABC, Generic[T]):
    @abstractmethod
    def a(self, arg: T) -> T:
        ...

    @abstractmethod
    def b(self, arg: T) -> T:
        ...

class Bar(Foo[int]):
    def a(self, arg: int) -> int:
        return arg

    def b(self, arg: str) -> str:
        return arg
```

The line at the end which defines `def b(self, arg: str) -> str:` has placed a `str` while the class `Bar` was defined with `Foo` associated with an `int`. This, to my knowledge, should imply all the `T`'s need to be of type `int`.

Running [mypy 1.5.1](https://github.com/python/mypy/releases/tag/v1.5.1) gives:
```
~$ mypy a.py
a.py:21: error: Return type "str" of "b" incompatible with return type "int" in supertype "Foo"  [override]
a.py:21: error: Argument 1 of "b" is incompatible with supertype "Foo"; supertype defines the argument type as "int"  [override]
a.py:21: note: This violates the Liskov substitution principle
a.py:21: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
```

While [pyright 1.1.323](https://github.com/microsoft/pyright/releases/tag/1.1.323) gives:
```
~$ pyright a.py
0 errors, 0 warnings, 0 informations
```

**Comment by erictraut** [comment]
Reporting of incompatible method overrides is not enabled by default in "basic" type checking. You need to enable the `reportIncompatibleMethodOverride` option (or enable "strict" mode, which implies this option and others).

You can enable `reportIncompatibleMethodOverride` globally in your project via the config file or locally to a single file by adding `# pyright: reportIncompatibleMethodOverride=true` to the top of the file.

For more information about configuration options, refer to [this documentation](https://microsoft.github.io/pyright/#/configuration?id=type-check-diagnostics-settings).

**Comment by egeres** [comment]
🤦🏻‍♂️ I'm so sorry I have reported a few non-bugs... I'll try to be more careful from now on, thank you for your work... 🙇🏻‍♂️
